modules/apps/client-extension/client-extension-web/src/main/resources/META-INF/resources/js/components/FileEntryPicker.tsx

```diff
@@ -0,0 +1,304 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import ClayAlert, {DisplayType} from '@clayui/alert';
+import ClayButton, {ClayButtonWithIcon} from '@clayui/button';
+import {Text} from '@clayui/core';
+import {ClayInput} from '@clayui/form';
+import ClayLoadingIndicator from '@clayui/loading-indicator';
+import {LearnMessage, LearnResourcesContext} from 'frontend-js-components-web';
+import {fetch, objectToFormData} from 'frontend-js-web';
+import PropTypes from 'prop-types';
+import React, {ChangeEvent, useEffect, useRef, useState} from 'react';
+
+type TFeedback = {displayType: DisplayType; message: string};
+
+const EMPTY_FEEDBACK: TFeedback = {
+	displayType: 'info',
+	message: '',
+};
+
+const disableFormSubmitButton = (
+	disabled: boolean,
+	portletNamespace: string
+) => {
+	const submitButton = document.getElementById(
+		portletNamespace + 'editClientExtensionEntrySubmitButton'
+	) as HTMLButtonElement;
+
+	if (submitButton) {
+		submitButton.disabled = disabled;
+	}
+};
+
+const readInputFile = (file: File) => {
+	const fileReader = new FileReader();
+
+	return new Promise<string>((resolve) => {
+		fileReader.onload = () => {
+			resolve(fileReader.result as string);
+		};
+
+		fileReader.readAsText(file);
+	});
+};
+
+const post = async (body: any, endpoint: string) => {
+	return await fetch('/o/frontend-token-definition' + endpoint, {
+		body,
```

Comment:
This component looks suitable to manage frontend token definitions file only, and not any file. So, its name should reflect that. `FrontendTokenDefinitionFilePicker` would likely be a more accurate name.

Reply:
It makes sense. Done!